.github/workflows/caching.yml

```diff
@@ -1,8 +1,8 @@
 name: Caching
 
 # Docs on the workflow:
-# 1. GitHub cache scoping goes: main branch -> PR main branch -> PR. (essentially in a tree fashion). That is why it is useful to build caches on `master` - to generate & keep the main project state & they would be shared to the whole tree.
-# 2. GitHub has a 10G default limit of cache pool per repo. HLS is a big project & monorepo of many projects, so to keep cache useful - the main branch state caches should be preserved & their storage preferred to the PRs, since PRs from internal branches - would count into 10G pool, but would be available only inside of the PR scope, which can bork cache fore the whole community. That is short story why `dist-newstyle` (especially full) - is not includded into `master` cache.
+# 1. GitHub cache scoping goes: main branch -> PR main branch -> PR. (essentially in a tree fashion). Building & keeping caches on `master` allows to the main project state cache be shared to the whole tree.
+# 2. GitHub has a 10G as default limit of cache pool per repo. HLS is a big community project & monorepo of many projects, so cache pool can get exhausted & invaidated easilly. Because of that, to keep cache useful - the main branch state caches should be preserved & kept well below the limit & that preferred to the PR caches, since PRs from internal branches - would count into the repo 10G pool, while PR cache is available only inside of the PR, which would bork cache fore the whole community. That is a short story why `dist-newstyle` (especially full) - is not includded into `master` cache.
```

Comment:
Those comments are gold, any chance to have them in multiple lines to help in their reading?

Reply:
Yes. It is useful to state something like this at the top of the workflow (or any design that is restricted by external things), to save people time, especially if they had some obvious ideas to improve caching while not knowing details of restrictions.

Ok. Would also add links there.